ph-ee-connector-ams-mifos/src/main/java/org/mifos/connector/ams/interop/AmsCommonService.java

```diff
@@ -129,6 +143,36 @@ public void repayLoan(Exchange e) {
         // cxfrsUtil.sendInOut("cxfrs:bean:ams.local.loan", e, headers, e.getIn().getBody());
     }
 
+    public String disburseLoan(String fineractTenantId, LoanDisbursementRequestDto loanDisbursementRequestDto, String channelRequest,
+            String basicAuthHeader) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+        headers.add("fineract-platform-tenantid", fineractTenantId);
+        headers.add("Authorization", basicAuthHeader);
+        ObjectMapper objectMapper = new ObjectMapper();
+        String requestBody;
+        TransactionChannelRequestDTO channelRequestDTO;
+        try {
+            requestBody = objectMapper.writeValueAsString(loanDisbursementRequestDto);
+            channelRequestDTO = objectMapper.readValue(channelRequest, TransactionChannelRequestDTO.class);
+        } catch (Exception ex) {
+            logger.info(ex.getMessage());
+            return null;
+        }
+        String accountId = channelRequestDTO.getPayer().getPartyIdInfo().getPartyIdentifier();
+        String url = amsInteropHostPath + amsInteropDisbursalTransactionPath.replace("{accountId}", accountId);
+        try {
+            RestTemplateUtil restTemplateUtil = new RestTemplateUtil();
+            ResponseEntity<String> exchange = restTemplateUtil.exchange(url, HttpMethod.POST, headers, requestBody);
+            logger.info(exchange.toString());
+            logger.info("Response: {} status: {}", exchange.getBody().toString(), exchange.getStatusCode());
+            return exchange.getBody();
+        } catch (Exception ex) {
+            logger.error(ex.getMessage());
+            return null;
```

Comment:
Don't return null it might cause a NPE

Reply:
I am required to return null because that would help me to know wheather an exception has been caused, i will set the zeebe variables accordingly. So, it would not cause NPE.